llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -6760,9 +6760,10 @@ void LoopVectorizationPlanner::plan(ElementCount UserVF, unsigned UserIC) {
 
 InstructionCost VPCostContext::getLegacyCost(Instruction *UI,
                                              ElementCount VF) const {
-  if (ForceTargetInstructionCost.getNumOccurrences())
+  InstructionCost Cost = CM.getInstructionCost(UI, VF);
+  if (Cost.isValid() && ForceTargetInstructionCost.getNumOccurrences())
     return InstructionCost(ForceTargetInstructionCost.getNumOccurrences());
-  return CM.getInstructionCost(UI, VF);
```

Comment:
This is not a fault with your patch, but it looks like it was already broken due to this:

```
  return InstructionCost(ForceTargetInstructionCost.getNumOccurrences());
```

I'm pretty sure it should be:

```
  return InstructionCost(ForceTargetInstructionCost);
```

similar to `LoopVectorizationCostModel::expectedCost`

Reply:
Yes, to test this we need a test that passes a large value, e.g. ` -force-target-instruction-cost=100` in `llvm/test/Transforms/LoopVectorize/AArch64/force-target-instruction-cost.ll` would do it